File: app/src/main/java/net/feheren_fekete/applist/applistpage/model/StartableData.java
package net.feheren_fekete.applist.applistpage.model;

import java.util.Comparator;

public abstract class StartableData extends BaseData {

    private String mName;

    public StartableData(long id, String name) {
        super(id);
        mName = name;
    }

    public String getName() {
        return mName;
    }

    public void setName(String name) {
        mName = name;
    }

    public abstract String getPackageName();

    public static final class NameComparator implements Comparator<StartableData> {
        @Override
        public int compare(StartableData lhs, StartableData rhs) {
            return lhs.getName().compareTo(rhs.getName());
        }
    }

}

File: app/src/main/java/net/feheren_fekete/applist/applistpage/model/StartableData.kt
package net.feheren_fekete.applist.applistpage.model

import java.util.Comparator

abstract class StartableData(id: Long,
                             val packageName: String,
                             var name: String,
                             var customName: String) : BaseData(id) {

    fun displayName() = if (customName.isNotEmpty()) customName else name

    class NameComparator : Comparator<StartableData> {
        override fun compare(lhs: StartableData, rhs: StartableData): Int {
            return lhs.displayName().toLowerCase().compareTo(rhs.displayName().toLowerCase())
        }
    }

}
